### PR TITLE
Fix language-specific printing of constants in JSON output

### PR DESCRIPTION
**Pull request by peterschrammel**

### CODE REVIEW

File: src/util/json_expr.cpp [hunk 1]
```diff
@@ -228,6 +228,27 @@ json_objectt json(
...
+        lang=std::unique_ptr<languaget>(get_default_language());
+    }
+
+    const typet &underlying_type=
+      type.id()==ID_c_bit_field?type.subtype():
+      type;
+
+    std::string type_string;
+    if(lang->from_type(underlying_type, type_string, ns))
+      assert(false && "unknown type");
```
Comment (romainbrenguier) [inline src/util/json_expr.cpp]: I think we shouldn't use assert anymore

**Comment by smowton** [comment]
Could we have a test to illustrate the kind of problem you're trying to overcome?

**Comment by peterschrammel** [comment]
@smowton, added a test case. 
The output is used in the trace debugger UI. With this fix `boolean`s will show up as `false`/`true` rather than `0`/`1`, `NULL` becomes `null`, etc. There's certainly more that remains to be fixed to make expr2java print valid Java rather than JOTO'ish things.

**Comment by smowton** [comment]
RE: the above, are you aware of test-gen's `expr2cleanjava`? I initially started altering `expr2java`, then resolved to keep expr2java GOTO+Java special operations (e.g. new), while expr2cleanjava did things like hide pointer operations and array structures

**Comment by peterschrammel** [comment]
I'd like to get rid of expr2clean*. I don't see any use for it. There should be a neutral expr2goto and then language-specific ones... But that's independent of this PR, which has immediate customer impact.

**Pull request merged**